src/boot_O2/loadfragment.c

```diff
@@ -29,54 +29,59 @@ void Load_Relocate(void* allocatedVRamAddr, OverlayRelocationSection* ovl, u32 v
 
         switch (reloc & 0x3F000000) {
             case 0x2000000:
-                /* R_MIPS_32
-                 * Handles 32-bit address relocation.  Used in things such as
-                 * jump tables.
-                 */
+                // R_MIPS_32
+                // Handles 32-bit address relocation, used for things such as jump tables.
+
+                // Check address is valid for relocation
                 if ((*relocDataP & 0xF000000) == 0) {
                     *relocDataP = (*relocDataP - vRamStart) + allocu32;
-                } else {
-                    if (gLoadLogSeverity >= 3) {}
+                } else if (gLoadLogSeverity >= 3) {
                 }
                 break;
+
             case 0x4000000:
-                /* R_MIPS_26
-                 * Handles 26-bit address relocation, used for jumps and jals
-                 */
+                // R_MIPS_26
+                // Handles 26-bit address relocation, used for jumps and jals.
+
                 *relocDataP =
                     (*relocDataP & 0xFC000000) |
                     (((allocu32 + ((((*relocDataP & 0x3FFFFFF) << 2) | 0x80000000) - vRamStart)) & 0xFFFFFFF) >> 2);
                 break;
+
             case 0x5000000:
-                /* R_MIPS_HI16
-                 * Handles relocation for a lui instruciton, store the reference to
-                 * the instruction, and will update it in the R_MIPS_LO16 section.
-                 */
+                // R_MIPS_HI16
+                // Handles relocation for a lui instruction, part 1.
+                // Store the reference to the instruction, and update it in the R_MIPS_LO16 section.
+
                 luiRefs[(*relocDataP >> 0x10) & 0x1F] = relocDataP;
                 luiVals[(*relocDataP >> 0x10) & 0x1F] = *relocDataP;
                 break;
+
             case 0x6000000:
-                /* R_MIPS_LO16
-                 * Updates the LUI instruction to reflect the relocated address.
-                 * The full address is calculated from the LUI and lo parts, and then updated.
-                 * if the lo part is negative, add 1 to the lui.
-                 */
+                // R_MIPS_LO16
+                // Handles relocation for a lui instruction, part 2.
+                // Updates the LUI instruction to reflect the relocated address.
+                // The full address is calculated from the LUI and lo parts, and then updated.
+                // If the lo part is negative, add 1 to the lui.
```

Comment:
It's probably truer to say that together these handle the halves of a hi/lo pair. The hi part is always a `lui`, the bottom part can be pretty much any other immediate instruction.

Reply:
Perhaps something like
```c
case 0x5000000:
                // R_MIPS_HI16
                // Handles relocation for a hi/lo pair, part 1.
                // Store the reference to the instruction, and update it in the R_MIPS_LO16 section.
...
case 0x6000000:
                // R_MIPS_LO16
                // Handles relocation for a hi/lo pair, part 2.
                // Updates the LUI instruction (hi) to reflect the relocated address.
                // The full address is calculated from the LUI and lo parts, and then updated.
                // If the lo part is negative, add 1 to the LUI.

Reply:
I think that's clearer, yes. Would also like @Thar0 's approval, though.

Reply:
Sorry I missed this, I think it looks good although strictly the LO instruction must be one with a signed immediate like `addiu`, due to the `If the lo part is negative, add 1 to the LUI.`. This would break if for example `ori` was relocated.